adversary/repeat.go

```diff
@@ -0,0 +1,64 @@
+package adversary
+
+import (
+	"github.com/filecoin-project/go-f3/gpbft"
+)
+
+type (
+	Repeat struct {
+		id            gpbft.ActorID
+		host          gpbft.Host
+		echoCountDist CountSampler
+	}
+	CountSampler interface {
+		// TODO replace with math/rand/v2 rand.Source once upgraded to go 1.22.
+		Uint64() uint64
+	}
+)
+
+func NewRepeat(id gpbft.ActorID, host gpbft.Host, echoCountDist CountSampler) *Repeat {
+	return &Repeat{
+		id:            id,
+		host:          host,
+		echoCountDist: echoCountDist,
+	}
+}
+
+func (r *Repeat) ReceiveMessage(msg *gpbft.GMessage, _ bool) (bool, error) {
```

Comment:
Can we assert the interface we're conforming to?

Reply:
Addressed in https://github.com/filecoin-project/go-f3/pull/164